src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx

```diff
@@ -74,16 +75,33 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
     purchaseHistoryRequestStatus: boolean;
   };
 
+  const futureOrOngoingVenueEvents =
+    venueEvents &&
+    venueEvents.filter(
+      (event: VenueEvent) =>
+        event.start_utc_seconds +
+          event.duration_minutes * ONE_MINUTE_IN_SECONDS >
+        Date.now() / 1000
+    );
+
   const { eventId, redirectTo } = getQueryParameters(location.search);
 
   venue && updateTheme(venue);
 
   useEffect(() => {
-    if (user && venueEvents && venueId && eventId && redirectTo === "payment") {
-      setSelectedEvent(venueEvents.find((event) => event.id === eventId));
+    if (
+      user &&
+      futureOrOngoingVenueEvents &&
+      venueId &&
+      eventId &&
+      redirectTo === "payment"
+    ) {
+      setSelectedEvent(
+        futureOrOngoingVenueEvents.find((event) => event.id === eventId)
```

Comment:
what happens if this is undefined?

Reply:
aye. A possible mitigation is to change its type to something non-nullable to prevent this being a problem.